inf-clojure.el

```diff
@@ -281,17 +281,17 @@ ALWAYS-ASK).  Otherwise get a list of all active inf-clojure
 REPLS and offer a choice.  It's recommended to rename REPL
 buffers after they are created with `rename-buffer'."
   (interactive "P")
-  (let ((new-repl-buffer
-         (if (and (not always-ask)
-                  (inf-clojure-repl-p))
-             (setq inf-clojure-buffer (current-buffer))
-           (let ((repl-buffers (inf-clojure-repls)))
-             (if (> (length repl-buffers) 0)
-                 (when-let ((repl-buffer (completing-read "Select default REPL: " repl-buffers nil t)))
-                   (setq inf-clojure-buffer (get-buffer repl-buffer)))
-               (user-error "No buffers have an inf-clojure process"))))))
-    (when new-repl-buffer
-      (message "Current inf-clojure REPL set to %s" new-repl-buffer))))
+  (if (and (not always-ask)
+           (inf-clojure-repl-p))
+      (progn
+        (setq inf-clojure-buffer (current-buffer))
+        (message "Current inf-clojure REPL set to %s" inf-clojure-buffer))
```

Comment:
To reduce this message duplication you can just return the new-repl buffer at the end and set it outside the `if`s with a single message there. You can even the the code looking for the new repl to be a separate private function, so this would read a bit better. I'd prefer such less imperative code structure.